src/vue3-app/src/stores/modal/actions.ts

```
@@ -0,0 +1,23 @@
+    openModal(this: ModalState, {id, title, message, customData, onConfirm, onCancel}: Modal) {
+        this.modals.push({id, title, message, customData, onConfirm, onCancel});
+    },
+    closeModal(this: ModalState, id: string) {
+        this.modals = this.modals.filter(modal => modal.id !== id);
+    },
+    clearModals(this: ModalState) {
+        this.modals = [];
+    },
+    confirm(this: ModalState, id: string, params?: any) {
+        const modal = this.modals.find(modal => modal.id === id);
+        if (modal && modal.onConfirm) modal.onConfirm(params || modal.customData);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)